ecms_base/modules/custom/ecms_api_recipient/tests/src/Unit/Form/EcmsApiRecipientConfigFormTest.php

```diff
@@ -0,0 +1,347 @@
+    $this->configForm->submitForm($form, $this->formState);
+  }
+
+  /**
+   * Test the submit form with an empty role.
+   */
+  public function testSubmitFormEntityException(): void {
+    $this->formState->expects($this->once())
+      ->method('getValue')
+      ->with('allowed_content_types')
+      ->willReturn(self::SELECTED_NODE_TYPES);
+
+    $settingsConfig = $this->createMock(Config::class);
+    $settingsConfig->expects($this->once())
+      ->method('set')
+      ->with('allowed_content_types', self::SELECTED_NODE_TYPES)
+      ->willReturnSelf();
+
+    $this->configFactory->expects($this->once())
+      ->method('getEditable')
+      ->with('ecms_api_recipient.settings')
+      ->willReturn($settingsConfig);
+
+    $roleEntity = $this->createMock(RoleInterface::class);
+    $roleEntity->expects($this->never())
+      ->method('getPermissions')
+      ->willReturn(self::EXISTING_PERMISSIONS);
+
+    $roleEntity->expects($this->never())
+      ->method('revokePermission')
+      ->willReturnSelf();
+
+    $roleEntity->expects($this->never())
+      ->method('grantPermission')
+      ->willReturnSelf();
+
+    $roleEntity->expects($this->never())
+      ->method('save')
+      ->willReturnSelf();
+
+    $roleStorage = $this->createMock(EntityStorageInterface::class);
+    $roleStorage->expects($this->once())
+      ->method('load')
+      ->with(self::RECIPIENT_ROLE)
+      ->willReturn(NULL);
+
+    $this->entityTypeManager->expects($this->once())
+      ->method('getStorage')
+      ->with('user_role')
+      ->willReturn($roleStorage);
+
+    $form = [];
+    $this->configForm->submitForm($form, $this->formState);
+  }
+
+  /**
+   * Test the submit form with a storage exception.
+   */
+  public function testSubmitFormEntityStorageException(): void {
+    $this->formState->expects($this->once())
+      ->method('getValue')
+      ->with('allowed_content_types')
+      ->willReturn(self::SELECTED_NODE_TYPES);
+
+    $settingsConfig = $this->createMock(Config::class);
+    $settingsConfig->expects($this->once())
+      ->method('set')
+      ->with('allowed_content_types', self::SELECTED_NODE_TYPES)
+      ->willReturnSelf();
+
+    $this->configFactory->expects($this->once())
+      ->method('getEditable')
+      ->with('ecms_api_recipient.settings')
+      ->willReturn($settingsConfig);
+
+    $roleEntity = $this->createMock(RoleInterface::class);
+    $roleEntity->expects($this->once())
+      ->method('getPermissions')
+      ->willReturn(self::EXISTING_PERMISSIONS);
+
+    $roleEntity->expects($this->exactly(count(self::EXISTING_PERMISSIONS)))
+      ->method('revokePermission')
+      ->willReturnSelf();
+
+    $roleEntity->expects($this->exactly(count(self::SELECTED_NODE_TYPES) * 2))
+      ->method('grantPermission')
+      ->willReturnSelf();
+
+    $exception = $this->createMock(EntityStorageException::class);
+    $roleEntity->expects($this->once())
+      ->method('save')
+      ->willThrowException($exception);
```

Comment:
Just trying to understand why this should throw a storage exception?

Reply:
In the code I have a couple of try/catch blocks targeting that specific exception (https://github.com/State-of-Rhode-Island-eCMS/ecms_profile/pull/22/files#diff-465823c28cbf7c0ed554fd591ffa871eR166). In the testing, I was simulating an exception mostly so I could get 100% code coverage.

Reply:
Ok thanks!